pkg/cloud-controller-manager/instance.go

```diff
@@ -3,42 +3,43 @@ package ccm
 import (
 	"context"
 	"net"
+	"sync"
 
 	ctlkubevirtv1 "github.com/harvester/harvester/pkg/generated/controllers/kubevirt.io/v1"
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	cloudprovider "k8s.io/cloud-provider"
 	"k8s.io/cloud-provider/api"
 	kubevirtv1 "kubevirt.io/api/core/v1"
 )
 
 type instanceManager struct {
-	vmClient  ctlkubevirtv1.VirtualMachineClient
-	vmiClient ctlkubevirtv1.VirtualMachineInstanceClient
-	namespace string
+	vmCache      ctlkubevirtv1.VirtualMachineCache
+	vmiCache     ctlkubevirtv1.VirtualMachineInstanceCache
+	nodeToVMName *sync.Map
+	namespace    string
 }
 
 func (i *instanceManager) InstanceExists(ctx context.Context, node *v1.Node) (bool, error) {
-	if _, err := i.vmClient.Get(i.namespace, node.Name, metav1.GetOptions{}); err != nil && !errors.IsNotFound(err) {
-		return false, err
-	} else if errors.IsNotFound(err) {
+	if _, err := i.getVM(node); err != nil {
```

Comment:
The original method was using `vmClient.Get` to get data from remote instead of from cache; now the `getVM` is checking data from `sync.Map`, it will have such race:
when the `sync.Map` is not synced fully in `OnVmiChanged`; the `InstanceExists` will not work  as expected.

Reply:
Good catch. Reverted vmCache/vmiCache to vmClient/vmiClient. Thanks.